src/ray/gcs/redis_async_context.h

```diff
@@ -0,0 +1,71 @@
+#ifndef RAY_GCS_REDIS_ASYNC_CONTEXT_H
+#define RAY_GCS_REDIS_ASYNC_CONTEXT_H
+
+#include <stdarg.h>
+#include <mutex>
+#include "ray/common/status.h"
+
+extern "C" {
+#include "ray/thirdparty/hiredis/async.h"
+#include "ray/thirdparty/hiredis/hiredis.h"
+}
+
+namespace ray {
+
+namespace gcs {
+
+/// \class RedisAsyncContext
+/// RedisAsyncContext class is a wrapper of hiredis `asyncRedisContext`, providing
+/// C++ style and thread-safe API.
+/// NOTE(micafan): We use a lock to make the `asyncRedisContext` commands thread-safe.
+/// All these commands only manipulate memory data and don't actually do any IO
+/// operations. So the perf impact of adding the lock should be minimum.
+class RedisAsyncContext {
+ public:
+  explicit RedisAsyncContext(redisAsyncContext *redis_async_context)
+      : redis_async_context_(redis_async_context) {
+    RAY_CHECK(redis_async_context_ != nullptr);
+  }
+
+  ~RedisAsyncContext() {
+    redisAsyncFree(redis_async_context_);
+    redis_async_context_ = nullptr;
+  }
+
+  /// Get the raw 'redisAsyncContext' pointer.
+  redisAsyncContext *GetRawRedisAsyncContext() {
```

Comment:
Can you move all method definitions to the `.cc` file?

Reply:
Sure. I am working on it.